atlasdb-cli/src/main/java/com/palantir/atlasdb/cli/command/KvsMigrationCommand.java

```diff
@@ -192,4 +247,26 @@ private KeyValueServiceMigrator getMigrator(AtlasDbServices fromServices, AtlasD
                 .batchSize(batchSize)
                 .build());
     }
+
+    private static <K> Optional<K> loadFromFileOrInline(
+            @Nullable File configFile, String fileConfigRoot, @Nullable String inline, Class<K> clazz) {
+        return loadFromFile(configFile, fileConfigRoot, clazz).or(() -> Optional.ofNullable(inline)
+                .map(rethrowIoExceptionAsUnchecked(config -> AtlasDbConfigs.loadFromString(config, null, clazz))));
+    }
+
+    private static <K> Optional<K> loadFromFile(@Nullable File configFile, String fileConfigRoot, Class<K> clazz) {
+        return Optional.ofNullable(configFile)
+                .map(rethrowIoExceptionAsUnchecked(file -> AtlasDbConfigs.load(file, fileConfigRoot, clazz)));
+    }
+
+    private static <F, T, K extends IOException> Function<F, T> rethrowIoExceptionAsUnchecked(
+            FunctionCheckedException<F, T, K> function) {
+        return f -> {
+            try {
+                return function.apply(f);
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
```

Comment:
This is correct... but should we be using a logsafe exception instead?

Reply:
Went with SafeRuntimeException, as SafeIoException is still checked, and not convinced we gain much from creating our own SafeUncheckedIoException in this case.

Reply:
yeah that's what I'd have gone with too